cpp/src/strings/search/find_multiple.cu

```diff
@@ -31,37 +33,32 @@ namespace cudf {
 namespace strings {
 namespace detail {
 std::unique_ptr<column> find_multiple(
-  strings_column_view const& strings,
+  strings_column_view const& input,
   strings_column_view const& targets,
   rmm::cuda_stream_view stream,
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource())
 {
-  auto strings_count = strings.size();
-  if (strings_count == 0) return make_empty_column(type_id::INT32);
-  auto targets_count = targets.size();
+  auto const strings_count = input.size();
+  auto const targets_count = targets.size();
   CUDF_EXPECTS(targets_count > 0, "Must include at least one search target");
   CUDF_EXPECTS(!targets.has_nulls(), "Search targets cannot contain null strings");
 
-  auto strings_column = column_device_view::create(strings.parent(), stream);
+  auto strings_column = column_device_view::create(input.parent(), stream);
   auto d_strings      = *strings_column;
   auto targets_column = column_device_view::create(targets.parent(), stream);
   auto d_targets      = *targets_column;
 
+  auto const total_count = strings_count * targets_count;
+
   // create output column
-  auto total_count  = strings_count * targets_count;
-  auto results      = make_numeric_column(data_type{type_id::INT32},
-                                     total_count,
-                                     rmm::device_buffer{0, stream, mr},
-                                     0,
-                                     stream,
-                                     mr);  // no nulls
-  auto results_view = results->mutable_view();
-  auto d_results    = results_view.data<int32_t>();
+  auto results = make_numeric_column(
+    data_type{type_id::INT32}, total_count, rmm::device_buffer{0, stream, mr}, 0, stream, mr);
+
   // fill output column with position values
   thrust::transform(rmm::exec_policy(stream),
                     thrust::make_counting_iterator<size_type>(0),
                     thrust::make_counting_iterator<size_type>(total_count),
-                    d_results,
+                    results->mutable_view().begin<int32_t>(),
```

Comment:
Typically I would recommend to use `template begin` because sometimes the compiler may complain a template function is used without `template` keyword.